### PR TITLE
dart:ffi beta

### PR DESCRIPTION
**Pull request by kwalrath**
Contributes to https://github.com/dart-lang/sdk/issues/38770

Staged: https://sz-www-2.firebaseapp.com/guides/libraries/c-interop

### CODE REVIEW

**Comment by kwalrath** [comment]
@mit-mit should we update https://flutter.dev/docs/development/platform-integration/c-interop as well?

/cc @sfshaza2

**Comment by mit-mit** [comment]
lgtm

fyi @dcharkes @mkustermann

**Comment by mit-mit** [comment]
> @mit-mit should we update https://flutter.dev/docs/development/platform-integration/c-interop as well?

Yes, that would be good

File: src/_guides/libraries/c-interop.md [hunk 1]
```diff
@@ -12,16 +12,11 @@ _FFI_ stands for [_foreign function interface._][FFI]
...
-and isn't complete yet.
-The API is likely to have breaking changes
-between now and its completion.
-To use dart:fii, you need a 2.6 prerelease of the Dart SDK.
-If you're developing a Flutter app, use the Flutter dev channel,
-as described in the [Flutter dart:ffi page.][binding]
-Otherwise, use the
-[Dart dev channel](/get-dart#about-release-channels-and-version-strings).
+{{site.alert.info}}
+  As of Dart 2.6, [dart:ffi is in beta.][ffi issue]
```
Comment (dcharkes) [inline src/_guides/libraries/c-interop.md]: We should make the wording consistent with the library info maybe, this is the wording shipped with D26.

```dart
/**
 * Foreign Function Interface for interoperability with the C programming language.
 *
 * **NOTE**: Dart:FFI is in technical preview. The overall feature is incomplete,
 * may contain issues, and breaking API changes are still expected.
 *
 * For further details, please see: https://dart.dev/server/c-interop
 *
 * {@category VM}
 */
```

Comment (dcharkes) [reply src/_guides/libraries/c-interop.md]: In `sdk/lib/ffi/ffi.dart`.

**Comment by kwalrath** [comment]
Updated. ptal

**Comment by mit-mit** [comment]
LGTM

File: src/_guides/libraries/c-interop.md (hunk 1 above)
Comment (kwalrath) [reply src/_guides/libraries/c-interop.md]: Fixing in #2067

Comment (kwalrath) [reply src/_guides/libraries/c-interop.md]: I've also updated https://github.com/flutter/website/pull/3192 to match. But we won't merge that PR until we confirm that flutter dev channel is using 2.6.

**Pull request merged**